### DISCUSSION TITLE
Connection to DBT

### DISCUSSION THREAD

**Discussion by thegunner157** [apache/airflow#38047] [General] [answered]
Hi, I’m having Airflow platform created on Kubernetes cluster and encountered issue that I couldn't figure out. I want to use dbt using cosmos provider. In dockerfile of platform, in "/usr/src/dbt" path, I installed synapse venv using command from https://docs.astronomer.io/learn/airflow-dbt#step-1-configure-your-astro-project . I also set up all needed dbt configuration (dbt profile exists and can be found from DAG). I also set up ExecutablePath to installed virtual env. However while running simple DBT model, job is failing due to PermissionError on executable path ("PermissionError: [Errno 13] Permission denied: '/usr/src/dbt/venv/bin' ). Granting full access to this path using chmod -R 777 /usr/src/dbt/venv/bin in Dockerfile also does not help and still permission error is throwing. Do you know what can I do to fix this problem? Or what may be a reason for that?

**Comment by potiuk** [comment, marked as answer]
I think you should to alk to astronomer team. Generally speaking - if you use Airflow image - you have to make sure you follow the docs to extend the image https://airflow.apache.org/docs/docker-stack/build.html where you will find a lot of examples. But crucial part of this is that airflow can run as arbitrary user  - and you need to make sure that user have all the access. While you can use `airflow` user - you can also (for redshift compatibilty) use arbitrary user which has `0` group set - so the folders and files have to have the right permissions for that `0` group. 

BTiW. I think your mjstake is that you try to execute `bin` folder - rather than the binary in this folder and that is your likely issue

**Reply by thegunner157** [reply]
Thanks a lot for an answer! I think it may be the reason :)
